src/v/cloud_storage/remote_partition.cc

```diff
@@ -1100,15 +1103,16 @@ remote_partition::aborted_transactions(offset_range offsets) {
         auto cursor = std::move(cur_res.value());
         co_await for_each_manifest(
           std::move(cursor),
-          [&offsets, &meta_to_materialize](
+          [&offsets, &meta_to_materialize, this](
             ssx::task_local_ptr<const partition_manifest> manifest) {
               for (auto it = manifest->segment_containing(offsets.begin);
                    it != manifest->end();
                    ++it) {
                   if (it->base_offset > offsets.end_rp) {
                       return ss::stop_iteration::yes;
                   }
-                  auto path = manifest->generate_segment_path(*it);
+                  auto path = manifest->generate_segment_path(
+                    *it, _manifest_view->path_provider());
```

Comment:
the `manifest` could be an `STM` manifest which has different path provider instance

Reply:
We discussed this offline. The intent here isn't for readers to think "we're passing path providers around because they can be different path providers" but instead think "we're passing path providers around because we need to generate paths" with the understanding that there is a singular path provider per partition.

I'll make the path provider non-copyable, hopefully that will help